pkg/services/ngalert/state/state.go

```diff
@@ -268,7 +268,7 @@ func resultError(state *State, rule *models.AlertRule, result eval.Result, logge
 		resultAlerting(state, rule, result, logger)
 		// This is a special case where Alerting and Pending should also have an error and reason
 		state.Error = result.Error
-		state.StateReason = "error"
+		state.StateReason = models.StateReasonError
```

Comment:
this one is arguably more important in the long run. `StateReasonError` == `"Error"` != `"error"`